Description: CVE-2026-32775 - check maxlen to be at least 1
 libexif through 0.6.25 has a flaw in decoding MakerNotes. If the
 exif_mnote_data_get_value function gets passed in a 0 size, the
 passed in-buffer would be overwritten due to an integer underflow
 when maxlen-- is executed on a zero value. This patch adds a guard
 check that maxlen is at least 1 before proceeding with the decrement.
Origin: upstream, https://github.com/libexif/libexif/commit/7df372e9d31d7c993a22b913c813a5f7ec4f3692
Author: Marcus Meissner <marcus@jet.franken.de>
Last-Update: 2026-03-17

--- a/libexif/canon/mnote-canon-entry.c
+++ b/libexif/canon/mnote-canon-entry.c
@@ -559,6 +559,8 @@ mnote_canon_entry_get_value (const MnoteCanonEntry *entry, unsigned int t, char
 
 	if (!entry) 
 		return NULL;
+	if (maxlen < 1)
+		return NULL;
 
 	data = entry->data;
 	size = entry->size;
--- a/libexif/fuji/mnote-fuji-entry.c
+++ b/libexif/fuji/mnote-fuji-entry.c
@@ -199,6 +199,7 @@ mnote_fuji_entry_get_value (MnoteFujiEntry *entry,
 	int i, j;
 
 	if (!entry) return (NULL);
+	if (maxlen < 1) return NULL;
 
 	memset (val, 0, maxlen);
 	maxlen--;
--- a/libexif/olympus/mnote-olympus-entry.c
+++ b/libexif/olympus/mnote-olympus-entry.c
@@ -284,6 +284,8 @@ mnote_olympus_entry_get_value (MnoteOlympusEntry *entry, char *v, unsigned int m
 
 	if (!entry)
 		return (NULL);
+	if (maxlen < 1)
+		return NULL;
 
 	memset (v, 0, maxlen);
 	maxlen--;
--- a/libexif/pentax/mnote-pentax-entry.c
+++ b/libexif/pentax/mnote-pentax-entry.c
@@ -315,6 +315,7 @@ mnote_pentax_entry_get_value (MnotePentaxEntry *entry,
 	int i = 0, j = 0;
 
 	if (!entry) return (NULL);
+	if (maxlen < 1) return (NULL);
 
 	memset (val, 0, maxlen);
 	maxlen--;
